tests/python/pants_test/reporting/test_reporting_integration.py

```diff
@@ -265,6 +268,70 @@ def test_zipkin_reporter_for_v2_engine(self):
         v2_span_name_part, trace
         ))
 
+  def run_mock_server(self, bin_name, arguments):
```

Comment:
These functions are awesome :)
I think this function in particular would benefit from a doc explanation, and some type hints for the arguments, something like:

```suggestion
  def run_mock_server(self, bin_name: str, arguments: List[str]):
    """
    Build and run a Rust mock server binary.
    Mainly used for remoting integration tests.
    It assumes that the server will print a line like `localhost:9999` to indicate its port.

    :param bin_name: The name of the binary that you want to run. It must be in the main engine's workspace.
    :param arguments: Passthrough arguments to pass to the binary.
    """
```

Even better, these tests could go into a mixin, called something like `class RemoteServerMockMixin()`, and test suites that needed it could just extend it to have these functions for free (as they don't have side effects). The mixin should probably live in `tests/python/pants_test/engine`. This might be overkill for this PR, so okay to leave it as a TODO.

```suggestion
  #TODO: Move these utilities into a reusable mixin, possibly in tests/python/pants_test/engine
  def run_mock_server(self, bin_name, arguments):
```

Reply:
About extracting this to its own class, as @illicitonion mentioned, I had rather keep them local to here until we need to use them from somewhere else. Did the other changes suggested in this comment.